### PR TITLE
Add intermediate asset documentation

### PR DESCRIPTION
**Pull request by amzn-mike**
Signed-off-by: amzn-mike <80125227+amzn-mike@users.noreply.github.com>

## Change summary

Added documentation for intermediate assets feature

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by jackalbe — APPROVED** [review]

File: content/docs/user-guide/assets/pipeline/intermediate-assets.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* In the ProcessJob function, process the file and output the result as usual.  Create a JobProduct and set `m_outputFlags = AssetBuilderSDK::ProductOutputFlags::IntermediateAsset`.  Add it to the response: `response.m_outputProducts.push_back(jobProduct)`.
+
+That's all that's required to output an intermediate asset to be consumed by another builder.  Builders can be chained together like this for as many processing steps as required.
+
+##### Limitations
+ * An output must be either an intermediate or a product asset, not both.  If you want to output both, you'll need to create 2 separate jobs in CreateJobs.
+ * Product assets cannot output for the Common platform
+ * Intermediate assets must output for the Common platform
+ * Due to the above limitations, if you want to output both types of products, your ProcessJob code will need to be careful to output ProductAssets for non-common platform jobs and IntermediateAssets for common platform jobs.
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/pipeline/intermediate-assets.md:24]: Minor edit.

```suggestion
* An output must be either an intermediate or a product asset, not both.  If you want to output both, you'll need to create 2 separate jobs in CreateJobs.
* Product assets cannot output for the Common platform.
* Intermediate assets must output for the Common platform.
* Due to the above limitations, if you want to output both types of products, your ProcessJob code must output ProductAssets for non-common platform jobs and IntermediateAssets for common platform jobs.
```

File: content/docs/user-guide/assets/pipeline/intermediate-assets.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+If you want to output platform-specific intermediates, the recommended method is to output all your intermediates with platform prefixes on the name, and then output an additional manifest file which simply references all the other outputs.  For example you might make a job which outputs `pc_myshader.bin`, `mac_myshader.bin`, and `myshader.shadermanifest`.  You could then create another builder to consume these using `*.shadermanifest` as the input pattern.
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/pipeline/intermediate-assets.md:27]: Minor edit.

```suggestion

If you want to output platform-specific intermediates, the recommended method is to output all your intermediates with platform prefixes on the name, and then output an additional manifest file which simply references all the other outputs. For example you might make a job which outputs `pc_myshader.bin`, `mac_myshader.bin`, and `myshader.shadermanifest`. You could then create another builder to consume these using `*.shadermanifest` as the input pattern.
```

**Review by micronAMZN — CHANGES_REQUESTED** [review]

**Review by micronAMZN — CHANGES_REQUESTED** [review]
We should add an entry for this topic in assets/pipeline/_index.md

**Comment by amzn-mike** [comment]
> We should add an entry for this topic in assets/pipeline/_index.md

Added

**Review by micronAMZN — APPROVED** [review]

**Pull request merged**